src/release/breaking-changes/win_lifecycle_process_function.md

```diff
@@ -0,0 +1,42 @@
+---
+title: Introduction of FlutterEngine::ProcessExternalWindowMessage
+description: External windows shall call ProcessExternalWindowMessage to be considered for application lifecycle events
+---
+
+## Summary
+
+In order to be included in the logic for app lifecycle on Windows, any
+external windows added to a Flutter app shall invoke
+`FlutterEngine::ProcessExternalWindowMessage` from its `WndProc` function.
+
+## Description of change
+
+Implementing application lifecycle on Windows involves listening for Window
+messages in order to update the lifecycle state. In order for additional
+non-Flutter windows to affect the lifecycle state, they must forward their
+window messages to `FlutterEngine::ProcessExternalWindowMessage` from their
+`WndProc` functions. This function returns an `std::optional<LRESULT>`, which
+is `std::nullopt` when the message is received, but not consumed. When the
+returned result has a value, the message has been consumed, and further
+processing in `WndProc` should cease.
+
+## Migration guide
+
+An example `WndProc` procedure that invokes
+`FlutterEngine::ProcessExternalWindowMessage` is provided below:
+
+```
+CALLBACK LRESULT WndProc(HWND hwnd, UINT msg, WPARAM wparam, LPARAM lparam) {

```

Comment:
What's the `CALLBACK` here? Should we remove that?

Reply:
It's a macro that defines the `__stdcall` calling convention for Win32 callbacks.

Reply:
Oh derp that's a win32 thing, gotcha: https://learn.microsoft.com/en-us/windows/win32/winprog/windows-data-types